src/effects/SSGI.tsx

```diff
@@ -0,0 +1,154 @@
+import * as THREE from 'three'
+import { useMemo, useEffect, forwardRef } from 'react'
+import { useThree } from '@react-three/fiber'
+import { Pass, EffectPass } from 'postprocessing'
+// @ts-ignore
+import { VelocityDepthNormalPass, SSGIEffect, TRAAEffect } from 'realism-effects'
+
+export interface SSGIOptions {
+  distance: number
+  thickness: number
+  autoThickness: boolean
+  maxRoughness: number
+  blend: number
+  denoiseIterations: number
+  denoiseKernel: number
+  denoiseDiffuse: number
+  denoiseSpecular: number
+  depthPhi: number
+  normalPhi: number
+  roughnessPhi: number
+  envBlur: number
+  importanceSampling: boolean
+  directLightMultiplier: number
+  steps: number
+  refineSteps: number
+  spp: number
+  resolutionScale: number
+  missedRays: boolean
+}
+
+class SSGIPass extends Pass {
+  readonly velocityDepthNormalPass: VelocityDepthNormalPass
+  readonly ssgiPass: EffectPass
+  readonly traaPass: EffectPass
+
+  constructor(scene: THREE.Scene, camera: THREE.Camera, options: SSGIOptions = {} as SSGIOptions) {
+    super('SSGIPass', scene, camera)
+
+    this.velocityDepthNormalPass = new VelocityDepthNormalPass(scene, camera)
+    this.ssgiPass = new EffectPass(camera, new SSGIEffect(scene, camera, this.velocityDepthNormalPass, options))
+    this.traaPass = new EffectPass(camera, new TRAAEffect(scene, camera, this.velocityDepthNormalPass))
```

Comment:
`velocityDepthNormalPass` should maybe be separable so it can be used for other effects like motion blur or HBAO.

Reply:
or, we glob these together in the pass you made and allow the user to enable each part specifically

Reply:
I think we could have a provider and then individual effects since the order/combination is variable. Motion blur is the only one you'd want after TRAA, which could become redundant since TRAA and reprojection typically blurs on its own.